ibm/service/kubernetes/resource_ibm_container_dedicated_host.go

```diff
@@ -0,0 +1,482 @@
+			},
+			"life_cycle": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"actual_state": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"desired_state": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"message": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"message_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"message_details": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"message_details_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"resources": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The resources of the dedicated host",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"capacity": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"memory_bytes": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"vcpu": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"consumed": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"memory_bytes": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+									"vcpu": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"workers": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The workers of the dedicated host",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"cluster_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"flavor": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"worker_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"worker_pool_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceIBMContainerDedicatedHostCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(conns.ClientSession).VpcContainerAPI()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	dedicatedHostAPI := client.DedicatedHost()
+	targetEnv := v2.ClusterTargetHeader{}
+
+	hostPoolID := d.Get("host_pool_id").(string)
+
+	params := v2.CreateDedicatedHostRequest{
+		Flavor:     d.Get("flavor").(string),
+		HostPoolID: hostPoolID,
+		Zone:       d.Get("zone").(string),
+	}
+
+	res, err := dedicatedHostAPI.CreateDedicatedHost(params, targetEnv)
```

Comment:
good catch thx. Host pool is associated with a resource group, which is inherited by the host. 
We have tried creating a host pool with a specific resource group, it looked ok.
I have added the resource group id to the host pool.
Please note that, because the GET doesn't return the resource group id, we can only use it in create.